api/src/main/java/marquez/db/DatasetVersionDao.java

```diff
@@ -211,10 +207,10 @@ WITH selected_dataset_versions AS (
           FROM selected_dataset_versions dv
           LEFT JOIN runs_input_mapping rim
                ON rim.dataset_version_uuid = dv.uuid
-      ), selected_dataset_version_events AS (
-          SELECT dv.uuid, dv.dataset_name, dv.namespace_name, dv.run_uuid, le.event_time, le.event
+      ), selected_dataset_version_facets AS (
+          SELECT dv.uuid, dv.dataset_name, dv.namespace_name, dv.run_uuid, df.lineage_event_time, df.facet
           FROM selected_dataset_version_runs dv
-          LEFT JOIN lineage_events le ON le.run_uuid = dv.run_uuid
+          LEFT JOIN dataset_facets_view df ON df.dataset_uuid = dv.dataset_uuid AND df.run_uuid = dv.run_uuid
```

Comment:
The `selected_dataset_version_facets` CTE is expensive to create because we need to join on the `runs_input_mapping` table to find all of the runs that read from or wrote to the selected dataset version. Rather than going through this extra hop, can we add the dataset version UUID to the facet table so we can simply query all facets that apply to a specific dataset version?

Reply:
Yeah ;-) I had similar thoughts on this but the counterargument in my head was to avoid doing endlessly more & more in a single PR. 

I think you're right and there won't be better moment to fix this.

Reply:
Yeah ;-) I had similar thoughts on this but the counterargument in my head was to avoid doing endlessly more & more in a single PR.

I think you're right and there won't be better moment to fix this.
I've included it in a second commit within this PR.